src/main/java/org/broadinstitute/hellbender/engine/LocalReadShard.java

```diff
@@ -64,6 +68,13 @@ public LocalReadShard(final SimpleInterval interval, final ReadsDataSource reads
         this(interval, interval, readsSource);
     }
 
+    /**
+     * Reads in this shard will be transformed before filtering
+     *
+     * @param transformer read transformer to apply before read filtering
+     */
+    public void setPreReadFilterTransformer(final ReadTransformer transformer) { preReadFilterTransformer = transformer; }
```

Comment:
Add unit tests to `LocalReadShardUnitTest` proving that `setPreReadFilterTransformer()` and `setPostReadFilterTransformer()` are respected.